week-1/implement/sum.js

```diff
@@ -0,0 +1,9 @@
+function sum(arr){
+    const nonNumerical = arr.filter(item => typeof item === "number" && !isNaN(item));
+     const sumAll = nonNumerical.reduce((a,c)=>a+c,0);
+    if(nonNumerical === 0){
```

Comment:
You're comparing an array against 0.
The condition will never be true.
It's ok to return sum=0 for an empty array. No need to add the special check anyway.

Reply:
yes without code works properly without special check